src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -53,227 +52,203 @@
-        } else {
-            if (!eventType.isDeletable()) {
-                halt(HttpStatusCodes.FORBIDDEN, "you must not delete this event type");
-            }
-
-            blockingEventTypes = getBlockingEventTypes(eventType);
-            if (blockingEventTypes.isEmpty()) {
-                // delete queries
-                List<EventQuery> queries = PersistenceAdapterImpl.getInstance().getEventQueries(eventTypeId);
-                EventQuery[] queryArray = new EventQuery[queries.size()];
-                if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(queries.toArray(queryArray))) {
-                    halt(HttpStatusCodes.ERROR, "could not delete corresponding queries of event type");
-                }
-
-                // delete attributes
-                List<TypeAttribute> attributes = PersistenceAdapterImpl.getInstance().getTypeAttributes(eventTypeId);
-                TypeAttribute[] attributeArray = new TypeAttribute[attributes.size()];
-                if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(attributes.toArray(attributeArray))) {
-                    halt(HttpStatusCodes.ERROR, "could not delete corresponding attributes of event type");
-                }
-
-                // delete events
-                List<Event> events = PersistenceAdapterImpl.getInstance().getEventsOfEventType(eventTypeId);
-                Event[] eventArray = new Event[events.size()];
-                if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(events.toArray(eventArray))) {
-                    halt(HttpStatusCodes.ERROR, "could not delete corresponding events of event type");
-                }
-
-                // delete mappings
-                List<EventEntityMapping> mappings = PersistenceAdapterImpl.getInstance().getEventEntityMappingsForEventType(eventTypeId);
-                EventEntityMapping[] mappingArray = new EventEntityMapping[mappings.size()];
-                if (!PersistenceAdapterImpl.getInstance().deleteArtifacts(mappings.toArray(mappingArray))) {
-                    halt(HttpStatusCodes.ERROR, "could not delete corresponding mappings of event type");
-                }
-
-                deleteMappingConditions(mappings);
-
-                // delete event type
-                if (!PersistenceAdapterImpl.getInstance().deleteArtifact(eventType,
-                        EventTypeEndpoint.getDeleteEventTypeUri(eventType.getId()))) {
-                    halt(HttpStatusCodes.ERROR, "could not delete event type");
-                }
-
-                response.status(HttpStatusCodes.SUCCESS);
-            } else {
-                response.status(HttpStatusCodes.ERROR);
-                return blockingEventTypes;
-            }
         }
-
-        response.body("");
-        endpointUtil.logSendingResponse(logger, request, response.status(), response.body());
-        return response.body();
+		if (!eventType.isDeletable()) {

```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') NullPointerException might be thrown as 'eventType' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS2259)